src/main/java/com/odysseusinc/athena/service/job/VocabularyUpdatesChecker.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ * Copyright 2019 Odysseus Data Services, inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Company: Odysseus Data Services, Inc.
+ * Product Owner/Architecture: Gregory Klebanov
+ * Authors: Alexandr Cumarav
+ * Created: July 27, 2019
+ *
+ */
+
+package com.odysseusinc.athena.service.job;
+
+
+import com.odysseusinc.athena.repositories.athena.NotificationRepository;
+import com.odysseusinc.athena.service.NotificationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.Map;
+
+
+@ConditionalOnProperty(name = "scheduled.vocabulary.checker")
+@Component
+public class VocabularyUpdatesChecker {
+
+    private final NotificationRepository notificationRepository;
+    private final NotificationService notificationService;
+
+    @Value("${scheduled.vocabulary.checker}")
+    private String param;
+
+    @Autowired
+    public VocabularyUpdatesChecker(NotificationRepository notificationRepository, NotificationService notificationService) {
+        this.notificationRepository = notificationRepository;
+        this.notificationService = notificationService;
+    }
+
+    @Scheduled(cron = "${scheduled.vocabulary.checker}")
+    public void triggerCheckVocabularyUpdates() {
+
+        notificationService.ensureVocabularyVersionAndCodeAreSet();
+        sendVocabularyUpdatesNotification();
+
+    }
+
+    public void sendVocabularyUpdatesNotification() {
```

Comment:
Same as for `NotificationServiceImpl`: this breaks layer bounds
I expect here nothing more than `notificationService.notifySubscribers()`

Reply:
well I want to make each user notification as a separate transaction - In order to do that I have to put a bit of logic into this class, I think it is justified as we have VocabularyUpdateChecker class. Otherwise it will be just wrapper which connect crone to one more class which contains 3 lines of code.

Reply:
But in other hand there is scheduler containing some business logic violating principles of SOLID.

Reply:
Ok, I have splittet into VocabularyUpdatesJob and VocabularyUpdatesChecker